docs/user/ppl/cmd/rex.rst

```diff
@@ -0,0 +1,199 @@
+=============
+rex
+=============
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Description
+============
+| The ``rex`` command extracts fields from a raw text field using regular expression named capture groups.
+
+Version
+=======
+3.3.0
+
+Syntax
+============
+rex [mode=<mode>] field=<field> <pattern> [max_match=<int>] [offset_field=<string>]
+
+* field: mandatory. The field must be a string field to extract data from.
+* pattern: mandatory string. The regular expression pattern with named capture groups used to extract new fields. Pattern must contain at least one named capture group using ``(?<name>pattern)`` syntax.
+* mode: optional. Either ``extract`` (default) or ``sed``. In extract mode, creates new fields from named capture groups. In sed mode, performs text substitution on the field.
+
+  **Sed mode syntax:**
```

Comment:
Please reformat the doc. https://github.com/RyanL1997/sql/blob/202a69e3338f97d99542b623744f1c4e9058df15/docs/user/ppl/cmd/rex.rst

Reply:
Yes, and could you review it again? 

Couple changes I made: 
- reformatted the doc
- add the description of return type into each example